lib/bounce_email.rb

```diff
@@ -212,15 +217,29 @@ def check_if_bounce(mail)
       false
     end
 
-    def get_original_mail(mail) #worked alright for me, for sure this as to be extended
-      parts = mail.body.to_s.split(/--- Below this line is a copy of the message.(\r\n\r\n|\n\n)/)
-      if parts.size > 1
-        ::Mail.new(parts.last)
-      elsif mail.parts
-        ::Mail.new(mail.parts[2].body)
+    def get_original_mail(mail) #worked alright for me, for sure this has to be extended
+      if mail.multipart?
+        ::Mail.new(mail.parts.last)
+      elsif i = index_of_original_message_delimiter(mail)
+        ::Mail.new(extract_original_message_after(mail, i))
+      else
+        nil
       end
     rescue => e
+      puts e
```

Comment:
You've got a point. But just using `puts` is not good enough, if you ask me. What are the possible options I see:

1. Use a configurable logging mechanism. Even if it's on by default, it should be possible to turn it off, or change it to a user's proc.
2. Actually raise errors here, again configurable. E.g. if in a Rails env, we could default to raising errors.
3. Reconsider why at all do we have this `rescue` clause and if it can be removed, or at least, if it's scope can be narrowed.

I think, however, that this change is out of the scope of your pull request and we can leave it unchanged.